libraries/abstractions/transport/utest/transport_secure_sockets_utest.c

```diff
@@ -0,0 +1,740 @@
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesSent );
+
+    bytesSent = SecureSocketsTransport_Send( &invalidNetworkContext, networkBuffer, 0 );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesSent );
+
+    bytesSent = SecureSocketsTransport_Send( &invalidNetworkContext, networkBuffer, BYTES_TO_SEND );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesSent );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Send returns an error when #SOCKETS_Send fails to send
+ * data over the network stack.
+ */
+void test_SecureSocketsTransport_Send_Network_Error( void )
+{
+    int32_t bytesSent;
+
+    /* Several errors can be returned from #SSL_get_error as mentioned here:
+     * https://www.openssl.org/docs/man1.1.1/man3/SSL_get_error.html */
+    SOCKETS_Send_ExpectAnyArgsAndReturn( SECURE_SOCKETS_READ_WRITE_ERROR );
+    bytesSent = SecureSocketsTransport_Send( &networkContext, networkBuffer, BYTES_TO_SEND );
+    TEST_ASSERT_EQUAL( SECURE_SOCKETS_READ_WRITE_ERROR, bytesSent );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test the happy path case when #SecureSocketsTransport_Send is able to send all bytes
+ * over the network stack successfully.
+ */
+void test_SecureSocketsTransport_Send_All_Bytes_Sent_Successfully( void )
+{
+    int32_t bytesSent;
+
+    SOCKETS_Send_ExpectAnyArgsAndReturn( BYTES_TO_SEND );
+    bytesSent = SecureSocketsTransport_Send( &networkContext, networkBuffer, BYTES_TO_SEND );
+    TEST_ASSERT_EQUAL( BYTES_TO_SEND, bytesSent );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test the happy path case when #SecureSocketsTransport_Send but bytes lass then BYTES_TO_SEND.
+ */
+void test_SecureSocketsTransport_Send_Bytes_Sent_Partially( void )
+{
+    int32_t bytesSent;
+
+    SOCKETS_Send_ExpectAnyArgsAndReturn( BYTES_TO_SEND - 1 );
+    bytesSent = SecureSocketsTransport_Send( &networkContext, networkBuffer, BYTES_TO_SEND );
+    TEST_ASSERT_EQUAL( BYTES_TO_SEND - 1, bytesSent );
+}
+
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Recv is able to return that 0 bytes are received
+ * from the network stack when passing any invalid parameters.
+ */
+void test_SecureSocketsTransport_Recv_Invalid_Params( void )
+{
+    int32_t bytesReceived;
+    NetworkContext_t invalidNetworkContext = { 0 };
+    invalidNetworkContext.tcpSocket = NULL;
+
+    bytesReceived = SecureSocketsTransport_Recv( NULL, networkBuffer, BYTES_TO_RECV );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesReceived );
+
+    bytesReceived = SecureSocketsTransport_Recv( &invalidNetworkContext, NULL, BYTES_TO_RECV );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesReceived );
+
+    bytesReceived = SecureSocketsTransport_Recv( &invalidNetworkContext, networkBuffer, 0 );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesReceived );
+
```

Comment:
```suggestion

    invalidNetworkContext.tcpSocket = SOCKETS_INVALID_SOCKET;
```

Reply:
fixed in 5c6afe0c590c76d8c90a6972c9798058702963f1

resolving this comment, feel free to let me know if need more work.